src/Files/File.class.st

```diff
@@ -373,21 +373,23 @@ File class >> sizeOrNil: id [
 	^ nil
 ]
 
-{ #category : #'primitives-file' }
-File class >> stdioDescriptorIsATTY [
-	^ (0 to: 2)
-		anySatisfy: [ :fdNum | 
-			| res |
-			res := self fileDescriptorType: fdNum.
-			#(1 4) includes: res ]
-]
-
 { #category : #'primitives-file' }
 File class >> stdioHandles [
 	<primitive: 'primitiveFileStdioHandles' module: 'FilePlugin' error: ec>
 	self primitiveFailed
 ]
 
+{ #category : #'primitives-file' }
+File class >> stdioIsAvailable [
+	"Answer a boolean indicating whether stdio is available on the current platform.
+	stdio is considered available if any one of the three files (stdin, stdout, stderr) is available."
+
+	^ (0 to: 2)
+		anySatisfy: [ :fdNum | | res |
+			res := self fileDescriptorType: fdNum.
+			res between: 1 and: 3 ]
```

Comment:
Shouldn't this be `res between: 1 and: 4`, as per the previous code and the comment in `File class >> #fileDescriptorType:` ?

Reply:
Hi Damien,

I don't think so. 

From my response to the mailing list:

> Pharo can't write to a cygwin terminal since it uses the Windows
> Read/WriteConsole() functions.

> cygwin terminals are actually written to through pipes, and we can get
> the pipe information, and thus determine it is a cygwin terminal.  But
> that's it for now.

Cheers,
Alistair